gpflow/training/monitor/monitor.py

```diff
@@ -0,0 +1,790 @@
+        if self._summary is not None:
+            self._summary = tf.summary.merge(
+                [self._summary, tf.summary.scalar('optimisation/global_step',
+                                                  global_step_tensor)])
+        return self
+
+    def flush(self):
+        """
+        Flushes the event file to disk.
+        This can be called to make sure that all pending events have been written to disk.
+        """
+        self._file_writer.flush()
+
+    def _eval_summary(self, context: MonitorContext, feed_dict: Optional[Dict]=None) -> None:
+        """
+        Evaluates the summary tensor and writes the result to the event file.
+        :param context: Monitor context
+        :param feed_dict: Input values dictionary to be provided to the `session.run`
+        when evaluating the summary tensor.
+        """
+
+        if self._summary is None:
+            raise RuntimeError('TensorBoard monitor task should set the Tensorflow.Summary object')
+
+        if context.tf_session is None:
+            raise RuntimeError('To run a TensorBoard monitor task the TF session object'
+                               ' must be provided when creating an instance of the Monitor')
+
+        summary = context.tf_session.run(self._summary, feed_dict=feed_dict)
+        self._file_writer.add_summary(summary)
+        if self._flush_immediately:
+            self.flush()
+
+
+class StandardTensorBoardTask(BaseTensorBoardTask):
+    """
+    Monitoring task that creates a sensible TensorBoard for a model.
+    It sends to the TensorBoard the likelihood value and the model parameters.
+
+    The user can specify a list of model parameters that should be sent to the TensorBoard. By
+    default all model parameters are sent. The user can choose to send only scalar parameters.
+
+    It is possible to provide an additional list of Tensorflow summary objects that will be
+    merged with the model parameters' summary.
+    """
+
+    def __init__(self, model: Model, event_path: str, only_scalars: bool = True,
+                 parameters: Optional[List[Parameter]] = None,
+                 additional_summaries: Optional[List[tf.Summary]] = None) -> None:
+        """
+        :param model: Model tensor
+        :param event_path: Path of the event file where the summary protocol buffers will
+        be written to.
+        :param only_scalars: Restricts the list of output parameters to scalars.
+        :param parameters: List of model parameters to send to TensorBoard. If not
+        provided all parameters will be sent to TensorBoard.
+        :param additional_summaries: List of additional summary objects to send to TensorBoard.
+        """
+        super().__init__(model, event_path)
+        all_summaries = additional_summaries or []
+        parameters = parameters or list(model.parameters)
+
+        # Add scalar parameters
+        all_summaries += [tf.summary.scalar(p.full_name, tf.reshape(p.constrained_tensor, []))
```

Comment:
From `full_name` docstring: "WARNING: WILL BE DEPRICATED SOON."

Reply:
Should I use `pathname` instead?